src/io/flutter/run/test/TestConfigProducer.java

```diff
@@ -47,6 +48,8 @@ private boolean setupForDartFile(TestConfig config, ConfigurationContext context
     final PubRoot root = PubRoot.forPsiFile(file);
     if (root == null) return false;
 
+    if (!FlutterModuleUtils.hasFlutterModule(file.getProject())) return false;
```

Comment:
Nice catch.  `setupForDirectory` only gives us the directory (no config context and no module) so I think we're stuck with project scope there but here you're totally right we should be more picky. 👍